packages/common/src/request.ts

```diff
@@ -17,34 +17,54 @@ export class RemoteServerError extends Error {
 }
 
 /**
+ * ```js
+ * import { hubApiRequest } from "@esri/hub-common";
+ * //
+ * hubApiRequest(
+ *   "/datasets",
+ *   requestOptions
+ * })
+ *   .then(response);
+ * ```
  * make a request to the Hub API
- * @param url URL to request
+ * @param route API route
  * @param requestOptions request options
  */
-export function hubRequest(url: string, requestOptions?: IHubRequestOptions) {
-  // TODO: use request() under the hood?
-  // requestOptions.params.f = null;
-  // return request(url, requestOptions);
-  // TODO: cast to JSONAPI document?
-  // .then(response => {
-  //   const { data, meta } = response;
-  //   return { data, meta } as Document;
-  // });
-  // TODO: base on requestOptions
-  const fetchFn = /* requestOptions.fetch || */ fetch;
-  return fetchFn(url, {
-    method: "POST", // TODO: get from requestOptions?
-    headers: {
-      // TODO: base on request requestOptions?
-      "Content-Type": "application/json"
-    }
-    // TODO: base on requestOptions.params?
-    // body: JSON.stringify(body)
-  }).then(resp => {
-    if (resp.ok) {
-      return resp.json();
-    } else {
-      throw new RemoteServerError(resp.statusText, url, resp.status);
-    }
+export function hubApiRequest(
+  route: string,
+  requestOptions?: IHubRequestOptions
+) {
+  // merge in default request options
+  const options: IHubRequestOptions = {
+    ...{
+      hubApiUrl: "https://opendata.arcgis.com/api/v3/",
+      httpMethod: "POST",
+      fetch
```

Comment:
Why package `fetch` inside `options`?  Couldn't you just do something like:

`const fetchMethod = requestOptions.fetch || fetch;`

Reply:
you'd have to at least do `const fetchMethod = (requestOptions && requestOptions.fetch) || fetch;`. I don't see how that's any better than the current code

Reply:
Just readability, but that just opinion, so I will leave it up to you.  I saw that the `options.fetch` was possibly being modified, so I assumed `options` was going to be passed on to another method rather than just accessed and executed in the current block.

Reply:
gotcha, I've made that change for readability's sake.